core/templates/pages/topics-and-skills-dashboard-page/modals/create-new-topic-modal.component.ts

```diff
@@ -80,6 +80,9 @@ export class CreateNewTopicModalComponent extends ConfirmOrCancelModal {
     if (!this.newlyCreatedTopic.urlFragment) {
       return;
     }
+    this.newlyCreatedTopic.urlFragment = this.newlyCreatedTopic.urlFragment
+      .trim()
+      .replace(/\s+/g, '-');
```

Comment:
Here, spaces are being converted to hyphens, which is fine, but are we also automatically converting uppercase letters to lowercase? If not, why are we specifically handling spaces? Shouldn't we leave it to the editor to fix the URL fragment after receiving the error message? Any thoughts?

Reply:
The reason we’re specifically handling spaces is because, in the current implementation, the editor throws an error when there are trailing spaces in the URL, but the error message doesn’t explicitly mention this issue. Based on a suggestion from @HardikGoyal2003 , we want to align the URL field’s behavior with how the name field works.
In the name field, multiple spaces between words are reduced to a single space, and any trailing spaces are trimmed automatically. For the URL field, since spaces are not valid characters, I convert them to hyphens (-). This ensures the URL remains valid without requiring the editor to manually address trailing or extra spaces after receiving an error message.
Regarding the conversion of uppercase letters to lowercase, I haven’t added this functionality yet. However, it’s something I can consider to ensure consistent URL formatting across the board, especially since some systems treat URL casing inconsistently.
The current approach aims to automatically handle spaces, making the user experience smoother and reducing potential errors without requiring the editor to intervene manually.

Reply:
should I do upper case to lower case automatically in this PR?

Reply:
There may be other cases to consider as invalid, such as scenarios where the editor might add numbers, punctuation, or other characters. Automatically converting every possible input to our accepted standard isn't always feasible.

That said, I’m not opposed to the approach of automatically converting space-separated words to hyphen-separated ones. Is this discussed in the issues thread?

If not, I suggest displaying an error message, something like: "The URL fragment cannot contain the character <character>." For spaces specifically, we could use: "Spaces are not allowed in the URL fragment."

Before starting implementation, it would be useful to think through different cases and determine the error messages you want to display. Once the approach is approved, you can proceed with implementation, saving time.

Let me know if you have any questions.

Thanks!

Reply:
Thank you for sharing, in my opinion having only a regex validation check that you are doing in the `else-if` block can be sufficient because it will cover the above `if` condition.

Any thoughts?

Reply:
According to the issue thread discussion and your above suggestion, I implemented conditional showing an error message. As you say, only  ```else-if```  block code is required, and that is already in the code base.
If anything required to refine then please inform me.

Reply:
@Nik-09 PTAL

Reply:
@Nik-09 
The input field restricts entries to the maximum allowable length, preventing any additional characters beyond that limit.

> Sorry, @Somnath-Mishra, for the delay. One final question: can you please ensure the "validation of length" as per Hardik's comment? [#20850 (comment)](https://github.com/oppia/oppia/issues/20850#issuecomment-2427237919)